test/accumulate_by_key-spec.js

```diff
@@ -221,3 +227,52 @@ describe('accumulate_by_key should', () => {
         expect(results.length).toBe(0);
     });
 });
+
+describe('accumulate should', () => {
+    const job = newTestJobConfig();
+
+    job.operations = [
+        {
+            _op: 'test-reader',
+            fetcher_data_file_path: path.join(__dirname, 'fixtures', 'data.json')
+        },
+        {
+            _op: 'accumulate_by_key',
+            key_field: 'id',
+            empty_after: 10,
+            flush_data_on_shutdown: true
+        }
+    ];
+
+    it('return accumulated data on flush event', async () => {
+        const testSlice = newTestSlice();
+
+        const harness = new WorkerTestHarness(job, { assetDir: path.join(__dirname, '..', 'asset') });
+        await harness.initialize();
+
+        let results = await harness.runSlice(testSlice);
+        expect(results.length).toBe(0);
+
+        results = await harness.flush();
+        await harness.shutdown();
+
+        expect(results.length).toBe(3);
+    });
+
+    it('return nothing on flush event', async () => {
+        const testSlice = newTestSlice();
+
+        job.operations[1].flush_data_on_shutdown = false;
```

Comment:
Modifying the `job` like this might cause problems for future tests because they would be dependent on this mutation.
```suggestion
        const jobWithFlush = cloneDeep(job);
        jobWithFlush.operations[1].flush_data_on_shutdown = false;
```

Make sure to use to import cloneDeep from `lodash` or `@terascope/job-components` and use new cloned job when constructing the `WorkerTestHarness` below.